src/parser/ir/ir_parser_test.cc

```diff
@@ -21,17 +21,17 @@
 
 namespace raksha::ir {
 
-class IrParserTest : public testing::TestWithParam<absl::string_view> {};
+class IrParserRoundtripTest : public testing::TestWithParam<absl::string_view> {};
```

Comment:
I do worry about relying upon round tripping `double`s. Unlike `int`s, it's relatively easy to write `double`s that cannot be exactly represented and end up with a round trip that does not work. I guess if we find some cases that work and stick to them, this could be ok. Not sure if there's anything to do for this comment, just wanted to note my unease.

Reply:
Yes, there's the normalizing tests for testing values that can't/don't roundtrip, this is just for testing with values that DO roundtrip.

I share your unease, but this was the best way I could think to handle both cases.